src/edu/hillel/lesson16/ValueCalculator.java

```diff
@@ -0,0 +1,55 @@
+package edu.hillel.lesson16;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static java.util.Arrays.stream;
+
+public class ValueCalculator {
+    private float[] array;
+    private static final int ARRAY_SIZE=1000000;
+    private static final int HALF_ARRAY_SIZE=ARRAY_SIZE/2;
+
+    public void method ()  {
+        long start = System.currentTimeMillis();
+
+        array=new float[ARRAY_SIZE];
+
+        for (int i = 0; i <ARRAY_SIZE ; i++) {
+            array[i]=1;
+        }
+        float[] arr1=new float[HALF_ARRAY_SIZE];
+        float[] arr2=new float[HALF_ARRAY_SIZE];
+
+        System.arraycopy(array,0,arr1,0,HALF_ARRAY_SIZE);
+        System.arraycopy(array,HALF_ARRAY_SIZE,arr2,0,HALF_ARRAY_SIZE);
+
+        ReCalcArray reCalcArray1 = new ReCalcArray(arr1);
+        Thread thread1= new Thread(reCalcArray1, "recalculation array 1");
+        thread1.start();
+
+        ReCalcArray reCalcArray2 = new ReCalcArray(arr2);
+        Thread thread2 = new Thread(reCalcArray2, "recalculation array 2");
+        thread2.start();
+
+        try {
+            thread1.join();
```

Comment:
Why did you use join here?

Reply:
I have done it for joining the thread and wait for it end in main thread. Than after ending of both threads I can get timings of running.
I decided to do so to get needed timing

Reply:
Yes, I see, it's correct